src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    @Test
+    void loginIsValid_lessThenSixCharactersLogin_notOk() {
+        user.setLogin("Login");
+        user.setPassword("ValidPassword");
+        user.setAge(25);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void loginIsValid_sixCharactersLogin_ok() {
+        user.setLogin("Login6");
+        user.setPassword("ValidPassword");
+        user.setAge(25);
+        assertTrue(registrationService.loginIsValid(user.getLogin()));
+        assertEquals(user.getLogin(),registrationService.register(user).getLogin());
+    }
+
+    @Test
+    void loginIsValid_moreThenSixCharactersLogin_ok() {
+        user.setLogin("UniqueLogin");
+        user.setPassword("ValidPassword");
+        user.setAge(25);
+        assertTrue(registrationService.loginIsValid(user.getLogin()));
+        assertEquals(user.getLogin(),registrationService.register(user).getLogin());
+    }
+
+    @Test
+    void passwordIsValid_nullPassword_notOk() {
+        user.setLogin("NullPassLogin");
+        user.setPassword(null);
+        user.setAge(25);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void passwordIsValid_emptyStringPassword_notOk() {
+        user.setLogin("EmptyPasseLogin");
+        user.setPassword("");
+        user.setAge(25);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void passwordIsValid_lessThenSixCharactersPassword_notOk() {
+        user.setLogin("ShortPassLogin");
+        user.setPassword("Pass");
+        user.setAge(25);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void passwordIsValid_sixCharactersPassword_ok() {
+        user.setLogin("SixPassLogin");
+        user.setPassword("ValPas");
+        user.setAge(25);
+        assertTrue(registrationService.passwordIsValid(user.getPassword()));
+        assertEquals(user.getPassword(),registrationService.register(user).getPassword());
+    }
+
+    @Test
+    void passwordIsValid_moreThenSixCharactersPassword_ok() {
+        user.setLogin("LongPassLogin");
+        user.setPassword("NewPassword");
+        user.setAge(25);
+        assertTrue(registrationService.passwordIsValid(user.getPassword()));
+        assertEquals(user.getPassword(),registrationService.register(user).getPassword());
+    }
+
+    @Test
+    void ageIsValid_nullAge_notOk() {
+        user.setLogin("NullAgeLogin");
+        user.setPassword("NewPassword");
+        user.setAge(null);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void ageIsValid_lessThenEighteenAge_notOk() {
+        user.setLogin("YoungAgeLogin");
+        user.setPassword("NewPassword");
+        user.setAge(15);

```

Comment:
```suggestion
        user.setAge(17);
```
it is better to use the maximum allowable data in test cases
Check  it everywhere